### DISCUSSION TITLE
controlling kitty with scripts

### DISCUSSION THREAD

**Discussion by zilahir** [kovidgoyal/kitty#8076] [Q&A] [answered]
Hi! 

I am trying to solve a though-to-be-simple thing, but I am failing, so any help fro the community would be much appreciated. 🙏

I am creating a new `kitty` session using:

```bash
kitty --instance-group=some-groum --config ~/.config/kitty/some-config.conf --title some-title --session ~/.config/kitty/some-session -o allow_remote_control=yes
```

the session file is:

```bash
launch --hold zsh -ilc "ff --height=100%"

focus
```

- launching the alias `ff`, focusing it
- the `ff` is an interactive shell (an `fzf`), i hit enter
- and i'd like the created `kitty` session to close.

however, it is not closing, in fact, i am getting the following error:

```bash
function fff() {
	aerospace list-windows --all | fzf --bind '<some-output>'
}

fff "$@"
```

I've tried all sort of stuff, including `kill`-ing the session with `PPID` but nothig really worked. 

How would you achieve this?

Thank you i advance!

**Comment by kovidgoyal** [comment]
Presumably because you have --hold in there. And if you are enabling remote control you dont need to use session just do

```
kitty -o allow_remote_control=y zsh -ilc "whatever"
```

**Comment by zilahir** [comment]
Thank for the reply! 

Yeah been trying this also, and you are right. 

The problem is that the new session window is remaining on the dock (`macos 15.1`)

**Comment by kovidgoyal** [comment, marked as answer]
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.macos_quit_when_last_window_closed

**Reply by zilahir** [reply]
oh jeez, indeed this was it. read through the config multiple times but managed to miss it. Thank you! 🙏

**Reply by zilahir** [reply]
just one more thing i was also unable to figure out myself:

when executing this in the shell:

```
aerospace list-windows --all | fzf --bind 'enter:execute(zsh -ilc "aerospace focus --window-id {1}")+abort'
```

immediately after the window is closed the focus goes back to the previous workplace. any hint would be a tremendous help!

**Comment by kovidgoyal** [comment]
kitty has no control over where the focus goes after it is closed that's
up to the OS/window manager. You should use some scripting of the OS to
control that.
